### PR TITLE
dm: update precheck document

### PR DESCRIPTION
**Pull request by lichunzhu**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Update dm precheck document about new behaviors.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): 
- https://github.com/pingcap/tiflow/pull/7332
- https://github.com/pingcap/tiflow/pull/7349

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-precheck.md [hunk 1]
```diff
@@ -52,8 +52,13 @@ tiup dmctl check-task ./task.yaml
...
 
     - 检查上游表是否设置了外键。TiDB 不支持外键，如果上游表设置了外键，则返回警告。
-    - （必须）检查字符集是否存在兼容性差异，详见 [TiDB 支持的字符集](/character-set-and-collation.md)。
-    - （必须）检查上游表中是否存在主键或唯一键约束（从 v1.0.7 版本引入）。
+    - 检查字符集是否存在兼容性差异，详见 [TiDB 支持的字符集](/character-set-and-collation.md)。
+    - 检查上游表中是否存在主键或唯一键约束（从 v1.0.7 版本引入）。
+
+    > **警告：**
+    >
+    > - 上游使用不兼容字符集时下游可以建表 utf8mb4 兼容同步，但不建议这样做。建议在上游使用兼容字符集，或者在下游使用与上游相同的字符集。
```
Comment (lance6716) [inline dm/dm-precheck.md:60]:
```suggestion
    > - 上游使用不兼容字符集时下游可以以 utf8mb4 字符集建表兼容同步，但不建议这样做。建议调整上游使用的字符集，或者在下游使用与上游相同的字符集。
```

**Review by lance6716 — APPROVED** [review]

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: dm/dm-precheck.md [hunk 2]
```diff
@@ -52,8 +52,13 @@ tiup dmctl check-task ./task.yaml
...
 
     - 检查上游表是否设置了外键。TiDB 不支持外键，如果上游表设置了外键，则返回警告。
-    - （必须）检查字符集是否存在兼容性差异，详见 [TiDB 支持的字符集](/character-set-and-collation.md)。
-    - （必须）检查上游表中是否存在主键或唯一键约束（从 v1.0.7 版本引入）。
+    - 检查字符集是否存在兼容性差异，详见 [TiDB 支持的字符集](/character-set-and-collation.md)。
+    - 检查上游表中是否存在主键或唯一键约束（从 v1.0.7 版本引入）。
+
+    > **警告：**
+    >
+    > - 上游使用不兼容字符集时下游可以以 utf8mb4 字符集建表兼容同步，但不建议这样做。建议调整上游使用的字符集，或者在下游使用与上游相同的字符集。
```
[the remark this answers is not in the document]
Comment (qiancai) [reply dm/dm-precheck.md:60]: "或者在下游使用与上游相同的字符集" 这里有点困惑，当上游使用不兼容的字符集时，下游没法使用与上游相同的字符集吧？

[the remark this answers is not in the document]
Comment (lichunzhu) [reply dm/dm-precheck.md:60]: 这样如何
```suggestion
    > - 上游使用不兼容的字符集时，下游可以使用 utf8mb4 字符集建表兼容同步，但不建议这样做。建议调整上游使用的字符集为下游支持的字符集。
```

**Review by qiancai — APPROVED** [review]
LGTM

**Pull request merged**